### ISSUE TITLE
3.4 breaks focus highliting

### ISSUE THREAD

**Issue by jambonsupreme** [godotengine/godot#52620]
### Godot version

v3.4.beta4.official.6a058cbf3

### System information

Windows 10

### Issue description

[#47280](https://github.com/godotengine/godot/pull/47280) separated hover from focus, but I was relying on custom_colors/font_color_hover to higlight the selected menu item when moving the selection with the d-pad. This no longer works in 3.4. The selection moves around just fine, but with no highlighting. The addition of a custom_colors/font_color_focus property would help me to achieve the correct behaviour again, altough it would require a manual intervention.

### Steps to reproduce

1) Add 2 or more buttons
2) Set "custom_colors/font_color_hover" to Color.red
3) Grab focus on one of the buttons
4) Move the selection with the keyboard / joystick arrows

In 3.3 the focused button is highlighted and the highlighting moves as you move the selection around, while in 3.4 the highlighting only appears while the mouse is hovering on the button. 

### Minimal reproduction project

[font_color_hover.zip](https://github.com/godotengine/godot/files/7150479/font_color_hover.zip)

**Comment by Calinou** [comment]
cc @jasonwinterpixel

**Comment by filipworksdev** [comment]
Hey guys.  Editor menu items do not change color when mouse hovers over them but only the underline shows.  I remember they used to highlight when mouse was over but I can't be sure. If that's new then it might be related to this change?

**Comment by pycbouh** [comment]
@filipworksdev Can you show what you mean exactly?

**Comment by filipworksdev** [comment]
@pycbouh Yes of course. See in the screen cap below my mouse is on Editor tool button but it doesn't light up, only the underline underneath it it does. Did it always behave like this? Unless of course it lights up but is incredibly faint white and looks no different than the regular editor tool tip button? 
![image](https://user-images.githubusercontent.com/36826319/139601634-eb04f621-7993-4e99-a500-c868a99f973e.png)

**Comment by pycbouh** [comment]
Looks about the same for me in 3.3.4

![image](https://user-images.githubusercontent.com/11782833/139604758-c57ca0f4-cd9f-4fcc-98ea-81ea61c2a5ab.png)
